cpu/esp_common/esp-wifi/esp_wifi_netdev.c

```diff
@@ -383,41 +383,82 @@ esp_err_t _esp_wifi_rx_cb(void *buffer, uint16_t len, void *eb)
+    "INVALID",                              /*0 */
+    "UNSPECIFIED",                          /*1 */
+    "AUTH_EXPIRE",                          /*2 */
+    "AUTH_LEAVE",                           /*3 */
+    "ASSOC_EXPIRE",                         /*4 */
+    "ASSOC_TOOMANY",                        /*5 */
+    "NOT_AUTHED",                           /*6 */
+    "NOT_ASSOCED",                          /*7 */
+    "ASSOC_LEAVE",                          /*8 */
+    "ASSOC_NOT_AUTHED",                     /*9 */
+    "DISASSOC_PWRCAP_BAD",                  /*10 */
+    "DISASSOC_SUPCHAN_BAD",                 /*11 */
+
+    "IE_INVALID",                           /*13 */
+    "MIC_FAILURE",                          /*14 */
+    "4WAY_HANDSHAKE_TIMEOUT",               /*15 */
+    "GROUP_KEY_UPDATE_TIMEOUT",             /*16 */
+    "IE_IN_4WAY_DIFFERS",                   /*17 */
+    "GROUP_CIPHER_INVALID",                 /*18 */
+    "PAIRWISE_CIPHER_INVALID",              /*19 */
+    "AKMP_INVALID",                         /*20 */
+    "UNSUPP_RSN_IE_VERSION",                /*21 */
+    "INVALID_RSN_IE_CAP",                   /*22 */
+    "802_1X_AUTH_FAILED",                   /*23 */
+    "CIPHER_SUITE_REJECTED",                /*24 */
+    "TDLS_PEER_UNREACHABLE",                /*25 */
+    "TDLS_UNSPECIFIED",                     /*26 */
+    "SSP_REQUESTED_DISASSOC",               /*27 */
+    "NO_SSP_ROAMING_AGREEMENT",             /*28 */
+    "BAD_CIPHER_OR_AKM",                    /*29 */
+    "NOT_AUTHORIZED_THIS_LOCATION",         /*30 */
+    "SERVICE_CHANGE_PERCLUDES_TS",          /*31 */
+    "UNSPECIFIED_QOS",                      /*32 */
+    "NOT_ENOUGH_BANDWIDTH",                 /*33 */
+    "MISSING_ACKS",                         /*34 */
+    "EXCEEDED_TXOP",                        /*35 */
+    "STA_LEAVING",                          /*36 */
+    "END_BA",                               /*37 */
+    "UNKNOWN_BA",                           /*38 */
+    "TIMEOUT",                              /*39 */
+
+    "PEER_INITIATED",                       /*46 */
+    "AP_INITIATED",                         /*47 */
+    "INVALID_FT_ACTION_FRAME_COUNT",        /*48 */
+    "INVALID_PMKID",                        /*49 */
+    "INVALID_MDE",                          /*50 */
+    "INVALID_FTE",                          /*51 */
+
+    "TRANSMISSION_LINK_ESTABLISH_FAILED",   /*67 */
+    "ALTERATIVE_CHANNEL_OCCUPIED",          /*68 */
+
+    "BEACON_TIMEOUT",                       /*200 */
+    "NO_AP_FOUND",                          /*201 */
+    "AUTH_FAIL",                            /*202 */
+    "ASSOC_FAIL",                           /*203 */
+    "HANDSHAKE_TIMEOUT"                     /*204 */
 };
+
+const char *get_esp_disc_reason(uint8_t code)
+{
+    uint8_t offset = 0;
+    uint8_t valid_reasons_len = sizeof(_valid_esp_wifi_disc_reasons);
+    valid_reasons_len = valid_reasons_len / sizeof(_valid_esp_wifi_disc_reasons)[0];
```

Comment:
```suggestion
    uint8_t valid_reasons_len = ARRAY_SIZE(_valid_esp_wifi_disc_reasons);
```
Where is `_valid_esp_wifi_disc_reasons` it declared? The PR is not compilable as is, neither for `esp32-wroom-32` nor for `esp8266-esp-12x` due this undeclared variable.